### PR TITLE
feat(cli): generate Json Schema

### PR DESCRIPTION
**Pull request by elcoosp**
# Why

For better IDE experience.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new CLI tool for generating JSON schemas.
	- Added a new job in the GitHub Actions workflow for schema generation.
	- New public modules added to expand library capabilities.

- **Bug Fixes**
	- Streamlined the `validate` function in the message module for improved performance.

- **Documentation**
	- Updated URLs in documentation comments for better clarity.

- **Chores**
	- Added a new `Cargo.toml` for the schema package with necessary dependencies.

### CODE REVIEW

**Comment by KeisukeYamashita** [comment]
Hello, thank you for raising this PR.
Let me review it over the next few days.

File: cli/json-schema/config.json [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+            {
+              "type": "null"
+            }
+          ]
+        }
+      }
+    },
+    "DescriptionEmpty": {
+      "description": "DescriptionEmpty represents the subject-empty rule.",
+      "type": "object",
```
[the remark this answers is not in the document]
Comment (elcoosp) [reply cli/json-schema/config.json:65]: I created another [branch](https://github.com/elcoosp/commitlint-rs/commits/feat/make-rule-macro/) that could fix this inconsistency by using [`make_rule!`](https://github.com/elcoosp/commitlint-rs/pull/1/files#diff-ae36572e69c35267539c5fb8a9b0ccfc0570c48bc649d0ee40799db9a30f12b8R298) macros to create each rule struct, with predefined docs (which can only be output to the json schema using [`schemars@v1-alpha.x`](https://docs.rs/schemars/1.0.0-alpha.16/schemars/index.html) if you are interested.

**Comment by KeisukeYamashita** [comment]
Thank you for your fantastic suggestion!
I'd like to not generate the schema on test because we either have to generate and then commit it or commit the generated schema on CI. Instead, I updated it so that the schema is added to each release.
Documentations will come soon.

File: cli/json-schema/config.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (KeisukeYamashita) [reply cli/json-schema/config.json]: Hello, in my humble opinion, I think it is out of this PR scope! 
Could you file another PR, please? Thank you in advance!

**Review by KeisukeYamashita — APPROVED** [review]

**Pull request merged**